### PR TITLE
NER model with ELMo

### PR DESCRIPTION
**Pull request by matt-peters**
Adding ELMo to the current NER model and adjusting hyperparameters.  ConLL 2003 test set F1 from one run was 92.53 compared to the reported value of 92.22 +/- 0.10 over 5 seeds from [the ELMo paper](https://arxiv.org/abs/1802.05365).

### CODE REVIEW

File: training_config/ner.json [hunk 1]
```diff
@@ -16,9 +16,9 @@
      }
     }
   },
-  "train_data_path": "/path/to/eng.train",
-  "validation_data_path": "/path/to/eng.testa",
-  "test_data_path": "/path/to/eng.testb",
+  "train_data_path": "${NER_TRAIN_DATA_PATH}",
+  "validation_data_path": "${NER_TEST_A_PATH}",
+  "test_data_path": "/${NER_TEST_B_PATH}",
```
Comment (matt-peters) [inline training_config/ner.json]: Thanks, good to know about the error message. Extra leading slash here

**Comment by DeNeutoy** [comment]
Looks great - thanks for doing this Matt, it would have taken me considerably longer to tweak all the params 👍

**Comment by joelgrus** [comment]
the one thing I would warn about (and it's possible you don't care) is that if you have `Parameter`s that only exist conditionally based on the config, you might run into trouble trying to make predictions from a trained model with a different config.

that is, if I train a model with start-end transitions enabled, and then want to predict without them, I think deep in the code `torch.load` will get angry that that there are tensors corresponding to (now) non-existent parameters, and vice versa the other way.

but it's possible that shouldn't even be allowed, in which case it doesn't matter.

(I ran into this issue when writing the "constraint mask" code of the CRF)

**Comment by DeNeutoy** [comment]
I don't think that's an issue we can really get around here - we have plenty of models with conditional components.

**Comment by schmmd** [comment]
89% -> 92.5% ;-)
